spec/controllers/admin_controller_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+describe AdminController do
+  describe '#index' do
+    it 'should be successful' do
+      get :index
+      expect(response).to be_successful
```

Comment:
Added tests for each type of user in `admin_controller_spec.rb` with `CanCan::AccessDenied` error assertions